crates/alpen-ee/common/src/traits/storage.rs

```diff
@@ -0,0 +1,96 @@
+use async_trait::async_trait;
+use strata_ee_acct_types::EeAccountState;
+use strata_identifiers::{OLBlockCommitment, OLBlockId};
+use thiserror::Error;
+
+use crate::EeAccountStateAtBlock;
+
+/// Identifies an OL block either by block ID or slot number.
+#[derive(Debug)]
+pub enum OLBlockOrSlot<'a> {
+    /// Identifies by block ID.
+    Block(&'a OLBlockId),
```

Comment:
Also, since OLBlockId is a `Copy` with just 32 bytes, we might not need this lifetime param here.